STEP_Project/OUListContent.js

```diff
@@ -31,9 +52,40 @@ function addButtons(OUs, numRows)
     }
 }
 
+/**
+ * This function implements Mutation Observer
+ * It checks if new OUs are visible and adds buttons to new rows if required
+ */
+function monitorChanges()
+{
+    let allRowNodes = document.querySelectorAll("tbody[role=rowgroup]");
+    let rootNode = allRowNodes[0];
+
+    let MutationObserver = window.MutationObserver;
+
+    let observer = new MutationObserver(function(mutations) 
+    {
+        let tabl = document.querySelector('table[role=grid]');
+        let OUs = tabl.rows;
+        let numOUs = OUs.length;
+
+        addButtons(OUs,numOUs);
```

Comment:
This part of code can be moved to a separate function say addButtonsToOrgUnitTable and the same can be reused here and above in initButtons.

Reply:
The code is slightly different in the _initButtons_ function because I need to handle an edge case (the number of rows in the two situations are different). If I create a function to do this, I'll also need to pass a boolean parameter to check whether or not the edge case needs to be handled, which might make it unnecessarily complex.